### DISCUSSION TITLE
Error spotbug en TPI2

### DISCUSSION THREAD

**Discussion by marciondg** [dds-jv/foro#249] [Consultas de herramientas] [answered]
Buenas tardes estimadxs,

Estoy teniendo este inconveniente a la hora de hacer el mvn verify y por consiguiente la entrega del TPI2:
![image](https://user-images.githubusercontent.com/53331055/120035377-e459e280-bfd4-11eb-9e2b-2752eb9279c7.png)

![image](https://user-images.githubusercontent.com/53331055/120035428-fb003980-bfd4-11eb-8347-d1981f95019a.png)

Busqué que la solución puede ser devolver una copia del array, probé con un Arrays.copyOf(bytes,getMaxSize()) pero rompen tests que antes funcionaban ok.

**Código error: EI_EXPOSE_REP**

Aguardo respuestas
Desde ya gracias
Saludos

**Comment by ciromartin** [comment, marked as answer]
Tuve el mismo problema: 

Lo resolvi modificando el archivo `spotbugs-exclude.xml`

Yo en mi caso particular la clase de Buffer la tenia en el package fs.files

En tu caso parece que tu clase Buffer esta en el package fs.highlevelfilesystem 

Modifica el tag Class con el package donde tenes la clase Buffer, es decir fs.highlevelfilesystem.Buffer

![image](https://user-images.githubusercontent.com/12116895/120048302-1e36e300-bfed-11eb-9716-806d84d8d4ea.png)

**Reply by nanderson87** [reply]
Efectivamente debería venir por ahí la solución, como dice @ciromartin . En el proyecto ya se había dejado el archivo .xml para omitir ese bug al hacer las verificaciones.

**Reply by marciondg** [reply]
Claro, la cagué agregando un package al pedo.
Muchas gracias a ambos!

**Comment by nubevi** [comment]
Buenas!

No se supone que el mio esta bien? Porque me tira error :/

![image](https://user-images.githubusercontent.com/26943847/120058090-ad102380-c01e-11eb-9c46-974fd963b3d7.png)

**Reply by nanderson87** [reply]
Te tira exactamente el mismo error? Nos podrás pasar una captura de pantalla del error cuando tiras el comando de maven?
